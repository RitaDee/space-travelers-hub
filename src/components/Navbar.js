import React from 'react';
import { NavLink } from 'react-router-dom';
import cn from 'classnames';
import logo from '../img/logo.png';

const links = [
  {
    to: '/',
    text: 'Rockets',
  },
  {
    to: '/missions',
    text: 'Missions',
  },
  {
    to: '/my-profile',
    text: 'My Profile',
  },
];

const Navbar = () => (

  <header className="header">
    <NavLink to="/" className="logos">
      <img src={logo} alt="planet" className="icon" />
      <span className="logo-text">Space Travelers&apos; Hub</span>
    </NavLink>

    <nav>
      <ul className="nav-list">
        {links.map((e) => (
          <NavLink
            key={e.to}
            to={e.to}
            className={({ isActive }) => cn('nav-link', { active: isActive })}
          >
            {e.text}
          </NavLink>

        ))}
      </ul>
    </nav>
  </header>
);

export default Navbar;
