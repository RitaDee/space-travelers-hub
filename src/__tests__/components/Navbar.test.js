import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from '../../components/Navbar';

describe('Navbar', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
