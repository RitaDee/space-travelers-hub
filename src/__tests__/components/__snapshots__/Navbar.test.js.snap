// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<header
  className="header"
>
  <a
    aria-current="page"
    className="logos active"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="planet"
      className="icon"
      src="logo.png"
    />
    <span
      className="logo-text"
    >
      Space Travelers' Hub
    </span>
  </a>
  <nav>
    <ul
      className="nav-list"
    >
      <a
        aria-current="page"
        className="nav-link active"
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
      <a
        className="nav-link"
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
      <a
        className="nav-link"
        href="/my-profile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </ul>
  </nav>
</header>
`;
